### PR TITLE
TSQL: `Create External Table`

### LINKED ISSUE #4628: TSQL: CREATE EXTERNAL TABLE unparsable
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

Microsoft SQL Server 2016 and later supports `CREATE EXTERNAL TABLE`, but is unparsable by sqlfluff.
[SQL Server Reference Documentation](https://learn.microsoft.com/en-us/sql/t-sql/statements/create-external-table-transact-sql?view=sql-server-2016&preserve-view=true&tabs=dedicated)

This example from the official [TSQL documentation](https://learn.microsoft.com/en-us/sql/t-sql/statements/create-external-table-transact-sql?view=sql-server-2016&preserve-view=true&tabs=dedicated#:~:text=G.%20Create%20an%20external%20table%20for%20SQL%20Server) is not parsable:
```
     CREATE EXTERNAL TABLE sqlserver.customer(
     C_CUSTKEY INT NOT NULL,
     C_NAME VARCHAR(25) NOT NULL,
     C_ADDRESS VARCHAR(40) NOT NULL,
     C_NATIONKEY INT NOT NULL,
     C_PHONE CHAR(15) NOT NULL,
     C_ACCTBAL DECIMAL(15,2) NOT NULL,
     C_MKTSEGMENT CHAR(10) NOT NULL,
     C_COMMENT VARCHAR(117) NOT NULL
      )
      WITH (
      LOCATION='tpch_10.dbo.customer',
      DATA_SOURCE=SqlServerInstance
     );

```

### Expected Behaviour

`CREATE EXTERNAL TABLE` should be parsable for tsql dialect 

### Observed Behaviour

`Found unparsable section`

### How to reproduce

Lint the following code snippet, using tsql dialect
```
     CREATE EXTERNAL TABLE sqlserver.customer(
     C_CUSTKEY INT NOT NULL,
     C_NAME VARCHAR(25) NOT NULL,
     C_ADDRESS VARCHAR(40) NOT NULL,
     C_NATIONKEY INT NOT NULL,
     C_PHONE CHAR(15) NOT NULL,
     C_ACCTBAL DECIMAL(15,2) NOT NULL,
     C_MKTSEGMENT CHAR(10) NOT NULL,
     C_COMMENT VARCHAR(117) NOT NULL
      )
      WITH (
      LOCATION='tpch_10.dbo.customer',
      DATA_SOURCE=SqlServerInstance
     );

```

### Dialect

tsql

### Version

2.0.2

### Configuration

N/A

### Are you willing to work on and submit a PR to address the issue?

- [x] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

### PR DESCRIPTION
**Pull request by aly76**
### Brief summary of the change made

fixes #4628 

### Pull Request checklist
- [x] Please confirm you have completed any of the necessary steps below.

- Included test cases to demonstrate any code changes, which may be one or more of the following:
  - `.yml` rule test cases in `test/fixtures/rules/std_rule_cases`.
  - `.sql`/`.yml` parser test cases in `test/fixtures/dialects` (note YML files can be auto generated with `tox -e generate-fixture-yml`).
  - Full autofix test cases in `test/fixtures/linter/autofix`.
  - Other.
- Added appropriate documentation for the change.
- Created GitHub issues for any relevant followup/future enhancements if appropriate.

### CODE REVIEW

File: src/sqlfluff/dialects/dialect_tsql.py [hunk 1]
```diff
@@ -5140,3 +5141,62 @@ class CreateExternalDataSourceStatementSegment(BaseSegment):
...
+    type = "create_external_table_statement"
+
+    match_grammar = Sequence(
+        "CREATE",
+        "EXTERNAL",
+        "TABLE",
+        Ref("ObjectReferenceSegment"),
+        Bracketed(
+            Delimited(
+                AnyNumberOf(Ref("ColumnDefinitionSegment")),
```
Comment (WittierDinosaur) [inline src/sqlfluff/dialects/dialect_tsql.py:5160]: You don't need this AnyNumberOf inside Delimited, because Delimited extends AnyNumberOf

**Comment by WittierDinosaur** [comment]
Barring my one comment, LGTM. Can you please run black, and fix the merge conflict?

**Comment by aly76** [comment]
> Barring my one comment, LGTM. Can you please run black, and fix the merge conflict?

Thanks for your feedback, I will get this fixed up!

**Review by WittierDinosaur — APPROVED** [review]
LGTM

**Pull request merged**